pkg/controller/attach_detach.go

```diff
@@ -251,6 +253,19 @@ func (r *ReconcileAttachDetach) triggerAttach(ctx context.Context, azVolumeAttac
 				_, uerr := updateState(azv, diskv1beta1.Attached, forceUpdate)
 				return uerr
 			}
+
+			if azVolumeAttachment.Spec.RequestedRole == diskv1beta1.PrimaryRole {
+				vaName, err := r.controllerSharedState.waitForVolumeAttachmentName(cloudCtx, azVolumeAttachment)
+				if err == nil {
+					vaUpdateFunc := func(obj interface{}) error {
+						va := obj.(*storagev1.VolumeAttachment)
+						va.Status.AttachmentMetadata = response
```

Comment:
It would be safer to simply add/replace an existing LUN entry rather than replace the entire map. Event though it's unlikely, there's no guarantee that I am aware of to prevent CSI from storing other values there in addition to that returned by ControllerPublishVolume.